### PR TITLE
Create test_eagerly_import.py

### PR DESCRIPTION
**Pull request by chuanzzx**
# Summary:
Add a test to check if Python imports are executed eagerly in default.

# Test Plan:
First of all, we need to check if it is in default (Lazy Imports is not enabled). If Lazy Imports is enabled, we do nothing in this test. On the other hand, if it is in default, we should check if imports are executed eagerly. Thus, this test is divided into two sections.
1. Check Lazy Imports is enabled or not
2. Check imports are executed eagerly in default

We create two classes to complete these two points. `LazyImportsEnabledChecker` is for the section one, and `TestEagerlyImport` is for the section two. Moreover, this test is wrapped by `unittest`, so we could just check the result and know if it works correctly.

## Check Lazy Imports is enabled or not
Two options make Lazy Imports work, and we check both of them in this test.
### 1. `-L` flag

If we add the `-L` or `-l` to our execution command, it should show _success_.
e.g.
```
./python.exe -mtest -v test_eagerly_import -L 
```
Expected result:
image

```
./python.exe -mtest -v test_eagerly_import -l
```
Expected result:
image

### 2. `PYTHONLAZYIMPORTSALL` environment variable
First, run this to set up the environment variable of Lazy Imports.
```
export PYTHONLAZYIMPORTSALL=1
```
Then, run this to check if it succeeds in without the flag.
```
./python.exe -mtest -v test_eagerly_import 
```
Expected result:
image

## Check imports are executed eagerly in default
We use `json` and `sqlite3` to confirm it. 
### json
`json` is not imported in the test code, so it and its sub-modules should not be loaded into `sys.modules`.
Thus, we check its sub-modules `json.encoder` and `json.decoder` in unit test to know if they are imported.

### sqlite3
After excluding all the edge cases, we could start to test if imports are really loaded when they are imported. We imported `sqlite3` at the beginning of the code (line 15), so its sub-module `sqlite3.dbapi2` should be in `sys.modules` even if we do not use it yet.

### Test behaviors in default
Open a new terminal (without setting `PYTHONLAZYIMPORTSALL`) and run 
```
./python.exe -mtest -v test_eagerly_import
```
Expected result:
image

### CODE REVIEW

File: Lib/test/test_eagerly_import.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        self.enabled = self.check_lazy_imports()
+
+    # confirm if Lazy Imports is enabled or not
+    def check_lazy_imports(self):
+        # lazy imports is not enabled in default
+        enabled = False
+
+        # check if Lazy Imports is turned on
+        enabled |= self.check_flag()
+        enabled |= self.check_env_var()
```
Comment (Kronuz) [inline Lib/test/test_eagerly_import.py:28]: It's probably not as relevant in this particular diff because these functions as not as expensive, but to avoid calling these methods once one has succeeded, you could have short-circuited like this:

```py
    return (
        self.check_flag() |
        self.check_env_var()
    )
```

This way `self.check_env_var()` doesn't get called if the first one returned `True`.

File: Lib/test/test_eagerly_import.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        return enabled
+
+    # check if the env variable PYTHONLAZYIMPORTSALL is set
+    def check_env_var(self):
+        env_variables = os.environ
+        if "PYTHONLAZYIMPORTSALL" not in env_variables:
+            return False
+
+        # check this setting of this env_var
+        if env_variables["PYTHONLAZYIMPORTSALL"] == "0":
```
Comment (Kronuz) [inline Lib/test/test_eagerly_import.py:38]: Oddly enough, setting `PYTHONLAZYIMPORTSALL` to `"0"` also enables Lazy Imports. Python's environment variables don't recognize anything but empty/non-empty strings. An empty string, however, would disable Lazy Imports.

File: Lib/test/test_eagerly_import.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            return False
+        else:
+            return True
+
+    # check if -L flag is added to the Python interpreter
+    def check_flag(self):
+        argvs = sys.argv
+        for argv in argvs:
+            if argv == "-L" or argv == "-l":
```
Comment (Kronuz) [inline Lib/test/test_eagerly_import.py:47]: Only uppercase `-L` enables lazy imports. Most likely, `-l` would result in an `Unknown option` error.

File: Lib/test/test_eagerly_import.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        self.modules = sys.modules
+
+    def test_json(self):
+        # we did not import json, so we should not have these sub-modules
+        self.assertFalse("json.encoder" in self.modules)
+        self.assertFalse("json.decoder" in self.modules)
+
+    def test_sqlite3(self):
+        # we import sqlite3 at the beginning, so we should have sqlite3.dbapi2 in sys.modules
+        self.assertTrue("sqlite3.dbapi2" in self.modules)
```
Comment (Kronuz) [inline Lib/test/test_eagerly_import.py:68]: While checking for keys in `sys.modules` may look like a viable solution in this isolated test, when running multiple tests this could result in flaky results. e.gr. Think for a moment some other test in the test suite imports (and uses) `json` or `sqlite3` during its tests, and that such test runs before our own test. In this cases the test would have failed.

What could be other way of testing this while ensuring this doesn't happen?

**Comment by Kronuz** [comment]
Other than the observations I left, this seems like a pretty good start @chuanzzx!

File: Lib/test/test_eagerly_import.py (hunk 4 above)
Comment (chuanzzx) [reply Lib/test/test_eagerly_import.py:68]: Is it better that we create two unique modules by ourselves and then replace `json` and `sqlite3` with them?
For example, we can create a module `Apple`, and a `Sub-Apple` is imported in `Apple`. Then, we could treat `Apple` as `json` and treat `Sub-Apple` as `json.encoder`. Does this make sense?

Comment (chuanzzx) [reply Lib/test/test_eagerly_import.py]: I created `imported_module` and `imported_module2` to replace `sqlite3`. Also, I imported them by using different ways. For `json`, I created `non_imported_module` to replace its original position.

**Comment by chuanzzx** [comment]
I modified the code based on the feedback. :) Also, I committed it and modified the test plan at the same time.

**Comment by chuanzzx** [comment]
Removed all .DS_Store files in this PR.

**Review by Kronuz — APPROVED** [review]
Thank you @chuanzzx! This looks good!!

**Comment by chuanzzx** [comment]
Yay! For the next step, should I merge this to your branch? Actually, I am not sure if I have this permission.
